codeserver/ubi9-python-3.12/test/test_startup.py

```
@@ -0,0 +1,96 @@
+            # (outdated?) https://github.com/opendatahub-io/odh-dashboard/blob/2.4.0-release/backend/src/utils/notebookUtils.ts#L284-L293
+            # https://github.com/opendatahub-io/odh-dashboard/blob/1d5a9065c10acc4706b84b06c67f27f16cf6dee7/frontend/src/api/k8s/notebooks.ts#L157-L170
+            "NOTEBOOK_ARGS": notebook_args,
+            # NB_PREFIX is set by notebook-controller and codeserver scripting depends on it
+            # https://github.com/opendatahub-io/kubeflow/blob/f924a96375988fe3801db883e99ce9ed1ab5939c/components/notebook-controller/controllers/notebook_controller.go#L417
+            "NB_PREFIX": nb_prefix,
+        }
+
+    def test_codeserver_startup(self):
+        env = self.get_environment_variables()
+        command = ["/opt/app-root/bin/run-code-server.sh"]
+        p = subprocess.Popen(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)